Automattic/woocommerce-payments
# Payments settings improvements

Closes https://github.com/Automattic/woocommerce-payments/issues/8246
The ACs for https://github.com/Automattic/woocommerce-payments/issues/8246 are essentially listed in https://github.com/Automattic/woocommerce-payments/issues/8246#issuecomment-1988771222.

#### Changes proposed in this Pull Request
This PR removes the spinner when checking the payment method on the Settings page (the spinner was there only to meet design expectations. Confirmed in [this discussion.](https://a8c.slack.com/archives/C01B8KNUYSW/p1710935261397159)

This PR also updates payment method statuses if they were updated after clicking the `Save settings` button. This means that merchants don't necessarily need to refresh the page to see the update.

#### Testing instructions

> Note: if you are able to set up your Stripe account in the way that it returns the capabilities you expect for testing, feel free to skip 1️⃣ and 2️⃣. For more inspiration, see https://github.com/Automattic/woocommerce-payments/issues/8246#issuecomment-1986347089

##### Payment methods checkbox has no spinner
* Enable any payment method in the settings page and confirm that the checkbox is enabled immediately without delay
* After enabling, save settings, refresh the page, and confirm that the payment method is still enabled
* Choose any payment method which has requirements to meet and confirm that after checking the checkbox, the activation modal window is enabled. Example:
    * disable `ideal` in Settings
    * 1️⃣ before [this line](https://github.com/Automattic/woocommerce-payments-server/blob/trunk/server/wp-content/rest-api-plugins/endpoints/wcpay/stripe/class-stripe-account.php#L306) in server, add `$this->account_data['capability_requirements']['ideal_payments'][] = 'business_profile.support_email';`, clear account cache
    * navigate to settings, check the iDEAL checkbox and confirm that the activation modal window is displayed

##### Payment methods list updates after saving settings without the need to refresh the page
* 2️⃣  modify [this line](https://github.com/Automattic/woocommerce-payments/pull/8428/files#diff-ca94d508b6d48cbc251483ca45da16a08991d4afd1d0f6c95c4e21baf398fdb2R566) with the following code
```php
		$res = $this->get_settings();
		$res->data['payment_method_statuses']['bancontact_payments'] = [
			'status' => 'rejected',
			'requirements' => 'business_profile.support_email'
		];
```
* navigate to settings, enable Bancontact, save settings and confirm that Bancontact status changed to `Rejected` without refreshing the page


-------------------


**Post merge**
